src/OmniSharp.Roslyn.CSharp/Services/Intellisense/IntellisenseService.cs

```diff
@@ -37,54 +37,60 @@ public IntellisenseService(OmniSharpWorkspace workspace, FormattingOptions forma
                         {
-                            // Note: For keywords, we'll just assume that the completion text is the same
-                            // as the display text.
-                            var keyword = item.DisplayText;
-                            if (keyword.IsValidCompletionFor(wordToComplete))
+                            var symbols = await item.GetCompletionSymbols(recommendedSymbols, document);
+                            if (symbols.Any())
                             {
-                                var response = new AutoCompleteResponse()
+                                foreach (var symbol in symbols)
                                 {
-                                    CompletionText = item.DisplayText,
-                                    DisplayText = item.DisplayText,
-                                    Snippet = item.DisplayText,
-                                    Kind = request.WantKind ? "Keyword" : null
-                                };
-
-                                completions.Add(response);
+                                    if (symbol != null)
+                                    {
+                                        if (request.WantSnippet)
+                                        {
+                                            foreach (var completion in MakeSnippetedResponses(request, symbol, item.DisplayText))
+                                            {
+                                                completions.Add(completion);
+                                            }
+                                        }
+                                        else
+                                        {
+                                            completions.Add(MakeAutoCompleteResponse(request, symbol, item.DisplayText));
+                                        }
+                                    }
+                                }
+
+                                // if we had any symbols from the completion, we can continue, otherwise it means
+                                // the completion didn't have an associated symbol so we'll add it manually
+                                continue;
                             }
-                        }
-                    }
-                }
 
-                var model = await document.GetSemanticModelAsync();
-                var symbols = await Recommender.GetRecommendedSymbolsAtPositionAsync(model, position, _workspace);
+                            // for other completions, i.e. keywords, create a simple AutoCompleteResponse
+                            // we'll just assume that the completion text is the same
+                            // as the display text.
+                            var response = new AutoCompleteResponse()
+                            {
+                                CompletionText = item.DisplayText,
```

Comment:
actually overrides and partials won't even reach that specific line because they'd branch out [here](https://github.com/OmniSharp/omnisharp-roslyn/pull/840/files#diff-d2949d6205937df4d863d016cb0be1f7R55) since these providers encode symbols. So the formatting/presentation of i.e. symbol information will be consistent with the rest of the experience, but yeah ultimately the `DisplayText` will be used for completion.

From what I experimented, even the simple `DisplayText` at the moment isn't that bad. It actually, in case of overrides, has the following format https://github.com/dotnet/roslyn/blob/614299ff83da9959fa07131c6d0ffbc58873b6ae/src/Features/Core/Portable/Completion/Providers/AbstractOverrideCompletionProvider.ItemGetter.cs#L21. So you get to see the full signature, and that's what can be inserted into the code even using the current object/endpoint model we have (of course things like missing usings would have to be resolved manually separately by the user)

It's definitely sub par experience compared to VS but at the same time maybe it's better than not having anything?
But of course, we could also just disable the complex providers like override/partial from participating in the old intellisense endpoint and only add them once we have a new one